saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncPersister.kt

```diff
@@ -86,7 +86,9 @@ class SavedSitesSyncPersister @Inject constructor(
     }
 
     private fun validateChanges(changes: SyncChangesResponse): SyncDataValidationResult<SyncBookmarkEntries> {
-        val response = kotlin.runCatching { Adapters.updatesAdapter.fromJson(changes.jsonString) }.getOrNull()
+        val response = kotlin.runCatching { Adapters.updatesAdapter.fromJson(changes.jsonString) }.getOrElse {
+            return SyncDataValidationResult.Error(reason = "Sync-Feature: JSON format incorrect, exception: $it")
+        }
 
         if (response == null) {
```

Comment:
As discussed offline, with this change the validation is no longer needed. You can remove it.